libnetwork/iptables/firewalld.go

```diff
@@ -159,13 +205,24 @@ func OnReloaded(callback func()) {
 }
 
 // Call some remote method to see whether the service is actually running.
-func checkRunning() bool {
+func checkRunning() (bool, error) {
+
+	var err error
 	if connection == nil {
-		return false
+		return false, nil
 	}
-	var zone string
-	err := connection.sysObj.Call(dbusInterface+".getDefaultZone", 0).Store(&zone)
-	return err == nil
+
+	ctx, cancel := context.WithTimeout(context.Background(), dbusConnectionTimeout)
+	defer cancel()
```

Comment:
The `checkRunning` function is going to need to accept a context argument -- see my other review comment. Conventionally, functions which accept contexts do not also apply their own timeout to the _entire_ operation; they leave that _policy_ decision up to the caller. (Emphasis on **entire** -- as a corollary, the function as the caller is responsible for setting the policy for its calls to the sub-steps of the operation it is implementing.) Pass a context which times out after `dbusConnectionTimeout` to the `checkRunning()` calls; don't derive the context inside the `checkRunning` function.